### PR TITLE
atoi

### PR DESCRIPTION
**Pull request by medymik**
recoding atoi that convert a string to an integer

### CODE REVIEW

File: conversions/c_atoi_str_to_integer.c [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#include <stdio.h>
```
Comment (kvedala) [inline conversions/c_atoi_str_to_integer.c:1]: please add documentation to your code in [doxygen](https://doxygen.nl) standards. You can also refer to [this file](https://github.com/kvedala/C/blob/master/numerical_methods/realtime_stats.c)

File: conversions/c_atoi_str_to_integer.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+// recoding the original atoi function in stdlib.h
+int	c_atoi(char *str)
+{
```
Comment (kvedala) [inline conversions/c_atoi_str_to_integer.c:5]: please follow a indentation and formatting standard - LLVM is what we are standardizing throughout i.e.,
* no tabs, no spaces
* indent width = 4 spaces

File: conversions/c_atoi_str_to_integer.c [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#include <stdio.h>
+
+// recoding the original atoi function in stdlib.h
+int	c_atoi(char *str)
```
Comment (kvedala) [inline conversions/c_atoi_str_to_integer.c:4]: shouldn't the input argument be a constant?

File: conversions/c_atoi_str_to_integer.c [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+		i++;
+	}
+	return (value);	
+}
+
+
+int	main()
+{
+	// Test
+	printf("%d\n", c_atoi("-234"));
```
Comment (kvedala) [inline conversions/c_atoi_str_to_integer.c:39]: `int main` should have a return value

File: conversions/c_atoi_str_to_integer.c (hunk 1 above)
Comment (medymik) [reply conversions/c_atoi_str_to_integer.c:8]: Done

File: conversions/c_atoi_str_to_integer.c (hunk 2 above)
Comment (medymik) [reply conversions/c_atoi_str_to_integer.c:18]: Done

File: conversions/c_atoi_str_to_integer.c (hunk 1 above)
Comment (kvedala) [reply conversions/c_atoi_str_to_integer.c:8]: The author name should be yours, dont you think 😄

File: conversions/c_atoi_str_to_integer.c [hunk 5]
```diff
@@ -1,40 +1,87 @@
...
-		value = value * 10 + sign * (str[i] - '0');
-		// manage the overflow
-		if (sign == 1 && prev > value)
-			return (-1);
-		else if (sign == -1 && prev < value)
-			return (0);
-		i++;
-	}
-	return (value);	
+    /**
```
Comment (kvedala) [inline conversions/c_atoi_str_to_integer.c:19]: the comment style -
```c
/**
 * something something
 */
```
is reserved and should not be using within a function body. Inside a function body, you should use the normal comment blocks `/* --- */`

File: conversions/c_atoi_str_to_integer.c [hunk 6]
```diff
@@ -1,40 +1,87 @@
...
+            return (0);
+        i++;
+    }
+    return (value);	
+}
+
+/**
+ * test the function implementation
+ */
+int test_c_atoi()
```
Comment (kvedala) [inline conversions/c_atoi_str_to_integer.c:61]: perfect!

File: conversions/c_atoi_str_to_integer.c (hunk 5 above)
Comment (medymik) [reply conversions/c_atoi_str_to_integer.c]: alright i used to follow other rules :)

File: conversions/c_atoi_str_to_integer.c [hunk 7]
```diff
@@ -0,0 +1,85 @@
+/**
+ * \file
+ * \brief Recoding the original atoi function in stdlib.h 
+<<<<<<< HEAD
```
Comment (kvedala) [inline conversions/c_atoi_str_to_integer.c:4]: merge conflict here not resolved properly.

Comment (medymik) [reply conversions/c_atoi_str_to_integer.c]: yes it was

**Comment by sandrotutberidze** [comment]
Atoi makes alphabetical character to integer

**Comment by kvedala** [comment]
@deadshotsb this can be merged

**Comment by deadshotsb** [comment]
@kvedala You should request a member access, I see that you are doing great.

**Comment by deadshotsb** [comment]
@medymik good work

**Comment by kvedala** [comment]
> @kvedala You should request a member access, I see that you are doing great.

I'd be glad to help out. However, this repo suffers from poor structuring and many bad and non-functioning code. 

I have tried to address them in [my fork](https://github.com/kvedala/C) and a corresponding pull request #543 which would help fix the issues in this repo. Otherwise, I intend to continue development on my branch which will auto-generate a well-[documented website](https://kvedala.github.io/C) as well.

I have not seen the original owners of this repo ever make a comment or a code in the last two months and if they request, I would be happy to join and help out.

**Pull request merged**